apps/cli/src/commands/boot-challenger.ts

```diff
@@ -114,26 +115,78 @@ const onAssertionConfirmedCb = async (nodeNum: any, commandInstance: Vorpal.Comm
+            commandInstance.log(`[${new Date().toISOString()}] Submitting a batch challenge for assertion ids ${assertionIds}.`);
+
+            try {
+                // Get the confirm data for all of the assertions
+                const [confirmData, confirmDataHash] = await getMultipleChallengeConfirmData(assertionIds);
+
+                // Set the confirm data to submit for a batch challenge
+                confirmDataToSubmit = confirmDataHash;
+
+            } catch (error) {
+                commandInstance.log(`[${new Date().toISOString()}] Error Getting Batch Confirm Data Error: Assertion Ids => ${assertionIds} ${(error as Error).message}`);
+                sendNotification(`Get Batch Confirm Data Error: Assertion Ids => ${assertionIds} ${(error as Error).message}`, commandInstance);
+                throw error;
+            }
+        }else{
+            commandInstance.log(`[${new Date().toISOString()}] Submitting a challenge for assertion id ${nodeNum}.`);
+
+            // Set the confirm data to submit for a single challenge
+            confirmDataToSubmit = assertionNode.confirmData;
         }
-        commandInstance.log(`[${new Date().toISOString()}] Submit Assertion Error: ${(error as Error).message}`);
-        sendNotification(`Submit Assertion Error: ${(error as Error).message}`, commandInstance);
-        throw error;
+
+        try {
+            // Submit a Batched Challenge
+            await submitAssertionToReferee(
+                cachedSecretKey,
+                nodeNum,
+                lastAssertionId,
+                confirmDataToSubmit,
+                assertionNode.createdAtBlock,
+                cachedSigner!.signer,
+            );
+
+            // Update the last submitted assertion time
+            lastAssertionTime = Date.now();
+
+            commandInstance.log(`[${new Date().toISOString()}] Submitted assertion: ${nodeNum}`);
+            } catch (error) {
+                if (error && (error as Error).message && (error as Error).message.includes('execution reverted: "9"')) {
+                    commandInstance.log(`[${new Date().toISOString()}] Could not submit challenge because it was already submitted`);
+                    lastAssertionTime = Date.now();
+                    return;
+                }
+                commandInstance.log(`[${new Date().toISOString()}] Submit Assertion Error: ${(error as Error).message}`);
+                sendNotification(`Submit Assertion Error: ${(error as Error).message}`, commandInstance);
+            }
+    }else{
+        // Log that the assertion was not submitted because it has not been enough time since the last assertion
+        commandInstance.log(`[${new Date().toISOString()}] Assertion ${nodeNum} not submitted because it has not been ${MINIMUM_TIME_BETWEEN_ASSERTIONS / (60 * 1000)} minutes since the last assertion.`);
     }
```

Comment:
Please consider code complexity, this else could return and the if above would not be required anymore.

Reply:
Thank you for pointing this out. I've not typically used that pattern in the past, but I agree, it is superior and I will try to default to it moving forward. 

Please point it out if you notice it in the future(I'm sure you will haha)! Thank you for helping me get better!